app/controllers/sessions_controller.rb

```diff
@@ -41,6 +37,7 @@ def callback
 
     handle_with(SessionsCallback, user_state: self,
       complete: lambda {
+        cookies.signed[:authentication_provider] = @handler_result.outputs[:authentication_provider]
```

Comment:
I think we need to only set this in certain cases.  I think this is good to call for the `when :returning_user, :new_password_user, :transferred_authentication` case.  For `:new_social_user`, the user isn't all the way created yet (as you can see on line 45 they are redirected to the social signup page).  I'd rather only set the `:authentication_provider` right before [this line](https://github.com/openstax/accounts/blob/spring-16-flow/app/controllers/signup_controller.rb#L25).  In that `success` block, the user will always only have one authentication, so you could just query it with `current_user.authentications.first.provider`.

Since it will be called in two places from two controllers, can you add a helper method in the base controller (can do in `/config/initializers/controllers.rb`) that takes the provider string and sets the cookie value?

Also, can you change the cookie key to `:last_signin_provider`?